instrumentation/struts-2/src/main/java/io/opentelemetry/javaagent/instrumentation/struts2/Struts2Tracer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.struts2;
+
+import com.opensymphony.xwork2.ActionInvocation;
+import com.opensymphony.xwork2.ActionProxy;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.servlet.ServletContextPath;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+
+public class Struts2Tracer extends BaseTracer {
+
+  public static final Struts2Tracer TRACER = new Struts2Tracer();
+
+  public Span startSpan(ActionInvocation actionInvocation) {
+    Object action = actionInvocation.getAction();
+    Class<?> actionClass = action.getClass();
+
+    String method = actionInvocation.getProxy().getMethod();
+    String spanName = spanNameForMethod(actionClass, method);
+
+    Span strutsSpan = tracer.spanBuilder(spanName).startSpan();
+
+    strutsSpan.setAttribute("code.namespace", actionClass.getName());
+    if (method != null) {
+      strutsSpan.setAttribute("code.function", method);
+    }
+
+    return strutsSpan;
+  }
+
+  // Handle cases where action parameters are encoded into URL path
+  public void updateServerSpanName(Context context, ActionProxy actionProxy) {
+    Span serverSpan = getCurrentServerSpan();
+    if (serverSpan == null) {
+      return;
+    }
+
+    // We take name from the config, because it contains the path pattern from the
+    // configuration.
+    String result = actionProxy.getConfig().getName();
+
+    String actionNamespace = actionProxy.getNamespace();
+    if (actionNamespace != null && !actionNamespace.isEmpty()) {
+      if (actionNamespace.endsWith("/") || result.startsWith("/")) {
+        result = actionNamespace + result;
+      } else {
+        result = actionNamespace + "/" + result;
+      }
+    }
+
+    if (!result.startsWith("/")) {
+      result = "/" + result;
+    }
+
+    if (!result.contains("{")) {
+      // If there are no braces, then there are no path parameters encoded in
+      // the action name, so let's not change existing server span name, because
+      // path is good enough. Wildcards like * in action name may glue
+      // several endpoints into one action name, which we do not want -- we want
+      // normalize parameters, not actions.
+      return;
+    }
+
+    getCurrentServerSpan().updateName(ServletContextPath.prepend(context, result));
+  }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.auto.struts-2";
```

Comment:
```suggestion
    return "io.opentelemetry.auto.struts-2.0";
```

Reply:
Renamed struts-2 to struts-2.3 everywhere.